include/eve/detail/function/simd/x86/load.hpp

```diff
@@ -11,125 +11,67 @@
 #pragma once
 
 #include <eve/as.hpp>
+#include <eve/concept/memory.hpp>
 #include <eve/concept/vectorizable.hpp>
 #include <eve/detail/abi.hpp>
-#include <eve/detail/meta.hpp>
-#include <eve/memory/aligned_ptr.hpp>
+#include <eve/detail/category.hpp>
 
 namespace eve::detail
 {
   //================================================================================================
-  // 128 bits loads
+  // Regular loads
   //================================================================================================
-  template<real_scalar_value T, typename N>
-  EVE_FORCEINLINE auto load(eve::as_<wide<T, N>> const &, eve::x86_128_ const &, T const* p)
+  template<scalar_pointer Ptr, typename T, typename N, x86_abi ABI>
+  EVE_FORCEINLINE auto load(eve::as_<wide<T, N>> const &, ABI const &, Ptr p)
```

Comment:
`as_<T, N, ABI>` ? Why didn't it fail tests?

Reply:
the fact the ABI is constrainted to x86_abi make it that wide<T,N> is the proper ABI.
I can add it for clarity though

Reply:
Ah - the test doesn't fail, because no-one ever specifies api themselves explicitly